libsolidity/ast/Types.cpp

```diff
@@ -669,7 +669,10 @@ tuple<bool, rational> RationalNumberType::isValidLiteral(Literal const& _literal
 			// parse the exponent
 			bigint exp = bigint(string(expPoint + 1, _literal.value().end()));
 
-			if (exp > numeric_limits<int32_t>::max() || exp < numeric_limits<int32_t>::min())
+			// SOL-008: Huge values for exponent are causing a crash. Limiting maximum to rational(1) << 4096
+			// 1<<1024 = 1.7977^308. Extrapolating for 1<<4096, the exponent value will approximately be 1232.
+			// However, it does appear that values in excess of E77 will still cause casting errors.
+			if (exp > 1232 || exp < -1232)
```

Comment:
In the end, this still depends on the mantissa. We could be conservative with the exponent, i.e. allow anything where `abs(exp) < 1250` and then re-check once we have the actual value. We should re-check this anyway at the end of the function because you can also do things like

`x = (1e20 * 1e20)** 500`.

The only reason we need a more intricate check at any point where `boost::multiprecision::pow` is used is that this operation itself could already exhaust memory.

Reply:
Going by the following error message:

```
biginttest.sol:6:11: Error: Type int_const 1000000000000000000000000000000000000000000000000000000000000000000000000000000 is not implicitly convertible to expected type uint256. c=1E78;
```

It appears that we are running into the hard limit of uint256 (2^256 = 1.157920892×10⁷⁷.) Exponent 77. This being the case, why not just set the upper limit to 77?  We are not going to be able to support anything bigger unless we start using a different data type.

Reply:
The idea is to allow intermediate values that are larger, as long as the final value of a computation fits into 256 bits.